packages/expo-background-task/src/BackgroundTask.ts

```diff
@@ -60,6 +60,13 @@ export async function registerTaskAsync(
       `Task '${taskName}' is not defined. You must define a task using TaskManager.defineTask before registering.`
     );
   }
+  if ((await ExpoBackgroundTaskModule.getStatusAsync()) === BackgroundTaskStatus.Restricted) {
```

Comment:
maybe we should warn instead of error, and just noop

Reply:
Good point!